script.rb

```diff
@@ -0,0 +1,228 @@
+      end
+      true
+    elsif block_given?
+      elements.my_each do |element|
+        return false unless yield element
+      end
+      true
+    elsif !block_given? && !arg
+      elements.my_each do |element|
+        return false unless element
+      end
+      true
+    elsif arg.class != Regexp && arg.class != Class
+      elements.my_each do |element|
+        return false unless element == arg
+      end
+      true
+    end
+  end
+
+  def my_any?(arg = nil)
+    elements = to_a
+    if arg == Numeric
+      elements.my_each do |element|
+        return true if element.class == Integer || element.class == Float
+      end
+      false
+    elsif arg.class == Class
+      elements.my_each do |element|
+        return true if element.class == arg
+      end
+      false
+    elsif arg.class == Regexp
+      elements.my_each do |element|
+        return true if element =~ arg
+      end
+      false
+    elsif block_given?
+      elements.my_each do |element|
+        return true if yield element
+      end
+      false
+    elsif !block_given? && !arg
+      elements.my_each do |element|
+        return true if element
+      end
+      false
+    elsif arg.class != Regexp && arg.class != Class
+      elements.my_each do |element|
+        return true if element == arg
+      end
+      false
+    end
+  end
+
+  def my_none?(arg = nil)
+    elements = to_a
+    if arg == Numeric
+      elements.my_each do |element|
+        return true unless element.class == Integer || element.class == Float
+      end
+      false
+    elsif arg.class == Class
+      elements.my_each do |element|
+        return false if element.class == arg
+      end
+      true
+    elsif arg.class == Regexp
+      elements.my_each do |element|
+        return false if element =~ arg
+      end
+      true
+    elsif block_given?
+      elements.my_each do |element|
+        return false if yield element
+      end
+      true
+    elsif !block_given? && !arg
+      elements.my_each do |element|
+        return false if element
+      end
+      true
+    elsif arg.class != Regexp && arg.class != Class
+      elements.my_each do |element|
+        return false if element == arg
+      end
+      true
+    end
+  end
+
+  def my_count(arg = nil)
+    elements = to_a
+    count = 0
+    if arg
+      elements.my_each do |element|
+        count += 1 if element == arg
+      end
+    elsif block_given?
+      elements.my_each do |element|
+        count += 1 if (yield element) == true
+      end
+    else
+      elements.my_each do |_element|
+        count += 1
+      end
+    end
+    count
+  end
+
+  def my_map(arg = nil)
+    return to_enum if !block_given? && !arg
+
+    elements = clone
+    elements.to_a
+    elements.my_each_with_index do |element, i|
+      elements[i] = if arg
+                      arg.call(element)
+                    else
+                      yield element
+                    end
+    end
+    elements
+  end
```

Comment:
- [ ] When `my_map` is called on a range, it should return a new array with the results of running a given block once for every element in enum.

```ruby
block = proc { |num| num < 4 }
range = Range.new(5, 50) 
range.my_map(&block) 
```

_Yours is currently returning a NoMethodError: `undefined method '[]=' for 5..50:Range`_